### PR TITLE
add reply feat and 404 page

### PR DESCRIPTION
**Pull request by hsnice16**
This PR will add code for the reply feature and 404 page.
Please review this PR and give your feedback.

### CODE REVIEW

**Review by Developers-codz — APPROVED** [review]
Well structured and readable code UI also looks awesome
Found nothing to suggest, you can merge

File: src/components/dialog/reply-dialog/ReplyDialog.jsx [hunk 1]
```diff
@@ -57,17 +60,34 @@ export const ReplyDialog = ({ openReplyDialog, setOpenReplyDialog }) => {
...
+  postUsername: PropTypes.string,
   openReplyDialog: PropTypes.bool,
+  postContentText: PropTypes.string,
+  postUserFullname: PropTypes.string,
   setOpenReplyDialog: PropTypes.func,
+  postUserProfilePic: PropTypes.string,
+  postTimeDurationToShow: PropTypes.string,
 };
 
 ReplyDialog.defaultProps = {
```
Comment (bhavyaj12) [inline src/components/dialog/reply-dialog/ReplyDialog.jsx:84]: Nice use of this

File: src/components/shared/avatar-grid-box/AvatarGridBox.jsx [hunk 2]
```diff
@@ -2,12 +2,19 @@ import PropTypes from "prop-types";
...
+  avatarSrc,
+  className,
+  handleClick,
+}) => {
   return (
     <Box
       gap={2}
       display="grid"
       className={className}
+      onClick={handleClick}
```
Comment (bhavyaj12) [inline src/components/shared/avatar-grid-box/AvatarGridBox.jsx:17]: handler name can be more descriptive

**Review by bhavyaj12 — APPROVED** [review]
Great work. Logic is extremely well written. File structure is great.
Good to merge.

File: src/components/shared/avatar-grid-box/AvatarGridBox.jsx (hunk 2 above)
Comment (hsnice16) [reply src/components/shared/avatar-grid-box/AvatarGridBox.jsx:17]: Thanks for suggesting @bhavyaj12 
I was trying to keep a generic name but will look once more if I can name something different and will add the changes in the future PR.

**Comment by hsnice16** [comment]
Thank you for reviewing,

**Pull request merged**